src/Core/ExecutionPathTracer/ExecutionPath.cs

```diff
@@ -0,0 +1,131 @@
+        /// </param>
+        public ExecutionPath(QubitDeclaration[] qubits, Operation[] operations)
+        {
+            this.Qubits = qubits;
+            this.Operations = operations;
+        }
+
+        /// <summary>
+        /// An array of <c>QubitDeclaration</c>s that represents the declared qubits used in the execution path.
+        /// </summary>
+        public QubitDeclaration[] Qubits { get; set; }
+
+        /// <summary>
+        /// An array of <c>Operation</c>s that represents the operations used in the execution path.
+        /// </summary>
+        public Operation[] Operations { get; set; }
+
+        /// <summary>
+        /// Serializes <c>ExecutionPath</c> into its JSON representation.
+        /// </summary>
+        /// <param name="prettyPrint">
+        /// Pretty prints the JSON (i.e. with white space and indents) if true (default = false).
+        /// </param>
+        public string ToJson(bool prettyPrint = false)
+        {
+            var options = new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                IgnoreNullValues = true,
+                WriteIndented = prettyPrint,
+            };
+            return JsonSerializer.Serialize(this, options);
+        }
+    }
+
+    /// <summary>
+    /// Represents a qubit resource used in an execution path.
+    /// </summary>
+    public class QubitDeclaration
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QubitDeclaration"/> class.
+        /// </summary>
+        /// <param name="id">
+        /// Id of qubit.
+        /// </param>
+        /// <param name="numChildren">
+        /// Number of associated classical registers.
+        /// </param>
+        public QubitDeclaration(int id, int? numChildren = null)
+        {
+            this.Id = id;
+            this.NumChildren = numChildren;
+        }
+
+        /// <summary>
+        /// Id of qubit.
+        /// </summary>
+        public int Id { get; set; }
+
+        /// <summary>
+        /// Number of associated classical registers.
+        /// </summary>
+        public int? NumChildren { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an operation used in an execution path.
+    /// </summary>
+    public class Operation
+    {
+        /// <summary>
+        /// Label of gate.
+        /// </summary>
+        public string Gate { get; set; }
+
+        /// <summary>
+        /// Non-qubit arguments provided to gate.
+        /// </summary>
+        public string ArgStr { get; set; }
+
+        /// <summary>
+        /// Group of operations for each classical branch.
+        /// </summary>
+        /// <remarks>
+        /// Currently not used as this is intended for classically-controlled operations.
+        /// </remarks>
+        public List<List<Operation>> Children { get; set; }
```

Comment:
Can this use `IEnumerable` rather than `List`? Also, similar comment as elsewhere about using `private set` wherever possible.

Reply:
I think in this case, since we don't have a constructor, we probably can't use `private set` here? Not sure what the convention is, but there would be a lot of arguments to pass into the construction for the `Operation` class, so I figured manually setting it would be the best way.